lib/traject/macros/bibtex.rb

```diff
@@ -0,0 +1,38 @@
+module Macros
+  # BibTeX extraction macros
+  module BibTeX
+    BIBTEX_ZOTERO_MAPPING = {
+      phdthesis: 'Thesis',
+      incollection: 'Book section',
+      article: 'Journal article',
+      book: 'Book',
+      misc: 'Document'
+    }.freeze
+
+    ##
+    # Traject macro for converting BibTeX fields into Solr fields
+    #
+    # @param [Symbol] `key` the BibTeX field name
+    def from_bibtex(key) # rubocop: disable Metrics/AbcSize, Metrics/CyclomaticComplexity
+      # Note that `record` is a BibTeX::Entry object
+      lambda do |record, accumulator, _context|
+        case key # BibTeX fields don't all follow the same rules, so handle them here
+        when :id # we need to transform the key for a good id
+          accumulator << record.key.to_s.gsub(%r{http:\/\/zotero.org\/groups\/\d*\/items\/}, '')
+        when :key # does not work with provides teste
+          accumulator << record.key.to_s
```

Comment:
@drh-stanford LGTM! One small thing: can you provide some additional context for this comment?

Reply:
ah yes. that is confusing :) i'll fix it